python/src/equistore/block.py

```diff
@@ -116,7 +116,9 @@ def __repr__(self) -> str:
         s += f"    samples ({len(self.samples)}): ["
         for sname in self.samples.names[:-1]:
             s += "'" + sname + "', "
-        s += "'" + self.samples.names[-1] + "']"
+        if len(self.samples.names) > 0:
+            s += "'" + self.samples.names[-1] + "'"
```

Comment:
Why are we only printing the last name here?

Reply:
yes the difference is that for all the other element we print `"self.samples.names" +","`. For the last one there is no “,“

Reply:
if `len(self.samples.names)==0` the loop is skipped